k9mail/src/main/java/com/fsck/k9/service/MailService.java

```diff
@@ -342,7 +344,10 @@ private void reschedulePushers(boolean hasConnectivity, boolean doBackground) {
 
     private void setupPushers() {
         boolean pushing = false;
-        for (Account account : Preferences.getPreferences(MailService.this).getAccounts()) {
+        List<Account> accounts =
+                new ArrayList<Account>(Preferences.getPreferences(MailService.this)
```

Comment:
This is fixing the problem at the wrong end. We need to make a defensive copy before returning the list in `Preferences.getAccounts()`